src/webgpu/api/operation/rendering/depth.spec.ts

```diff
@@ -3,13 +3,211 @@ Test related to depth buffer, depth op, compare func, etc.
+      format: renderTargetFormat,
+      size: { width: 1, height: 1, depthOrArrayLayers: 1 },
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.RENDER_ATTACHMENT,
+    });
+
+    const depthSpencilFormat: GPUTextureFormat = 'depth24plus-stencil8';
+    const depthTexture = this.device.createTexture({
+      size: { width: 1, height: 1, depthOrArrayLayers: 1 },
+      format: depthSpencilFormat,
+      sampleCount: 1,
+      mipLevelCount: 1,
+      usage: GPUTextureUsage.RENDER_ATTACHMENT | GPUTextureUsage.COPY_DST,
+    });
+
+    const depthStencilAttachment: GPURenderPassDepthStencilAttachment = {
+      view: depthTexture.createView(),
+      depthLoadOp: 'load',
+      depthStoreOp: 'store',
+    };
+
+    depthStencilAttachment.stencilLoadOp = 'load';
+    depthStencilAttachment.stencilStoreOp = 'store';
+
+    const encoder = this.device.createCommandEncoder();
+    const pass = encoder.beginRenderPass({
+      colorAttachments: [
+        {
+          view: renderTarget.createView(),
+          storeOp: 'store',
+          loadOp: 'load',
+        },
+      ],
+      depthStencilAttachment,
+    });
+
+    // Draw the base triangle with stencil reference 1.
+    {
+      const kBaseColor = new Float32Array([255, 255, 255, 255]);
+      const testPipeline = this.createRenderPipelineForTest(baseState, 0.0);
+      pass.setPipeline(testPipeline);
+      pass.setStencilReference(1);
+      pass.setBindGroup(
+        0,
+        this.createBindGroupForTest(testPipeline.getBindGroupLayout(0), kBaseColor)
+      );
+      pass.draw(6);
+    }
+
+    // Draw a triangle with the given stencil reference and the comparison function.
+    {
+      const testPipeline = this.createRenderPipelineForTest(state, 0.0);
+      pass.setPipeline(testPipeline);
+      pass.setStencilReference(stencilRefValue);
+      pass.setBindGroup(
+        0,
+        this.createBindGroupForTest(
+          testPipeline.getBindGroupLayout(0),
+          new Float32Array(stencilColor)
+        )
+      );
+      pass.draw(6);
+    }
+
+    pass.end();
+    this.device.queue.submit([encoder.finish()]);
+
+    this.expectSinglePixelIn2DTexture(
+      renderTarget,
+      renderTargetFormat,
+      { x: 0, y: 0 },
+      { exp: new Uint8Array(stencilColor) }
+    );
+  }
+
+  createRenderPipelineForTest(
+    depthStencil: GPUDepthStencilState,
+    depth: number
+  ): GPURenderPipeline {
+    return this.device.createRenderPipeline({
+      layout: 'auto',
+      fragment: {
+        targets: [{ format: 'rgba8unorm' }],
+        module: this.device.createShaderModule({
+          code: `
+            struct Params {
+                color : vec3<f32>,
+                depth : f32,
+            }
+            @group(0) @binding(0) var<uniform> params : Params;
+
+            @fragment fn main() -> @location(0) vec4<f32> {
+                return vec4<f32>(params.color, 1.0);
+            }`,
+        }),
+        entryPoint: 'main',
+      },
```

Comment:
nit: please order `vertex` before `fragment` in this descriptor. That's the order they run in, so it's a bit easier to read through the shader code that way.

Reply:
Ah, I see. Ordered `vertex` before `fragment`.